deploy-tool/main.go

```diff
@@ -338,6 +387,61 @@ func backupKeys(
 	return nil
 }
 
+func createManifests(
+	k8sSecretsClientGetter k8scorev1.SecretsGetter,
+	specificManifests map[string]SpecificManifestWrapper,
+	manifestFetcher ManifestFetcher,
+	manifestWriter manifest.DataShareProcessorSpecificManifestWriter,
+) error {
+	// Iterate over all specific manifests described by TF output so we can
+	// record any packet encryption key CSRs that have already been created. We
+	// assume that the presence of a packet encryption key CSR in any specific
+	// manifest means that a corresponding private key already exists in
+	// Kubernetes secrets.
+	existingManifests := map[string]struct{}{}
+	packetEncryptionKeyCSRs := manifest.PacketEncryptionKeyCSRs{}
+	for dataShareProcessorName := range specificManifests {
+		manifest, err := manifestFetcher.Fetch(dataShareProcessorName)
+		if err != nil {
+			return err
+		}
+
+		if manifest == nil {
+			continue
+		}
+
+		existingManifests[dataShareProcessorName] = struct{}{}
+
+		for keyName, certificateSigningRequest := range manifest.PacketEncryptionKeyCSRs {
```

Comment:
I understand the preconditions expected here, but ISTM we should verify that if `packetEncryptionKeyCSRs[keyName]` is not empty here, that it has the same CSR rather than clobbering it indiscriminately. 

This obviously can't happen when there are only two processors in the first place, but it'd be a painful bug if there were >2. Maybe it could just confirm that the key is empty for right now, and when this gets modified to handle additional processors as it might well be, it'll fail at that point and we can address it properly then.